src/lib/test_util/test_util.ml

```diff
@@ -46,6 +46,10 @@ module Make (Impl : Snarky.Snark_intf.S) = struct
     let checked_result = checked_to_unchecked typ1 typ2 checked input in
     assert (equal checked_result (unchecked input))
 
+  let arbitrary_string =
+    String.init (Random.int 10000000) ~f:(fun _ ->
```

Comment:
allocating a 10MB string one character at a time seems like it would be pretty slow -- do you want to move this down to 100K or even 10K?